File: leet_BST_0938_rangesum___bfs.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function rangeSumBST(root: TreeNode | null, low: number, high: number): number {
    let dq = [root]
    let a = []
    let temp: TreeNode | null
    let n: number
    while (dq.length) {
        temp = dq.shift()
        n = temp.val
        if (n <= high && n >= low) {
            a.push(n)
        }
        if (temp.left) {
            dq.push(temp.left)
        }
        if (temp.right) {
            dq.push(temp.right)
        }
    }
    return a.reduce((t, n) => t + n);
};

File: leet_BST_0938_rangesum___bfs.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func rangeSumBST(root *TreeNode, low int, high int) int {
    dq := [] *TreeNode { root }
    a := [] int {}
    for len(dq) != 0 {
        temp := dq[0]
        dq = dq[1:]
        n := temp.Val
        if n <= high && n >= low {
            a = append(a, n)
        }
        if temp.Left != nil {
            dq = append(dq, temp.Left)
        }
        if temp.Right != nil {
            dq = append(dq, temp.Right)
        }
    }
    res := 0
    for _, n := range a {
        res += n
    }
    return res
}
